src/FakeItEasy/Configuration/IPropertySetterConfiguration.cs

```diff
@@ -0,0 +1,19 @@
+namespace FakeItEasy.Configuration
+{
+    /// <summary>
+    /// Provides configuration for property setters and allows the user to specify validations for arguments.
+    /// </summary>
+    /// <typeparam name="TValue">The value of the property.</typeparam>
+    public interface IPropertySetterConfiguration<TValue>
+        : IVoidArgumentValidationConfiguration
+    {
+        /// <summary>
+        /// Configures the property assignment to be accepted when the value <see cref="object.Equals(object)"/>
+        /// <paramref name="value"/>.
+        /// </summary>
+        /// <param name="value">The value to match.</param>
+        /// <returns>A configuration object.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1716:IdentifiersShouldNotMatchKeywords", MessageId = "To", Justification = "There's no need for clients to implement the member.")]
+        IVoidArgumentValidationConfiguration To(TValue value);

```

Comment:
Shouldn't there be an overload that takes a lambda? I think we mentioned that at some point in #175.